Helper_Functions.py

```diff
@@ -0,0 +1,52 @@
+from PyPDF2 import PdfReader
+from langchain.text_splitter import CharacterTextSplitter
+from InstructorEmbedding import INSTRUCTOR
+from langchain.embeddings import HuggingFaceInstructEmbeddings
+from langchain.vectorstores import Milvus
+import spacy
+
+
+def get_pdf_text(file_path):
+    text = ""
+    with open(file_path, 'rb') as file:
+        pdf_reader = PdfReader(file)
+        for page in pdf_reader.pages:
+            text += page.extract_text()
+    return text
+
```

Comment:
def preprocess_text(text_chunks):
    text = text.lower()
    return text
  Preprocess shpould be done before text chunking otherwise it gives reference error